src/EventSubscriber/ParamConverterSubscriber.php

```diff
@@ -48,18 +53,69 @@ public function convert(JsonPreExecuteEvent $event): void
         $reflection = $event->getObjectReflection();
         $methodHandler = $event->getObject();
 
-        foreach ($params as $name => $value) {
-            if (!$reflection->hasProperty($name)) {
-                if ($this->parameterBag->get('rpc.server.parameters.allow_extra_params')) {
+        if (null !== $this->validator) {
+            $errors = [];
+            foreach ($params as $name => $value) {
+                if (!$reflection->hasProperty($name)) {
+                    if ($this->parameterBag->get('rpc.server.parameters.allow_extra_params')) {
+                        continue;
+                    }
+
+                    throw new InvalidParamsException(null, $jsonRequest->getId());
+                }
```

Comment:
I will add this check to both foreach, because we would skip it otherwise without validator

Reply:
The validator is almost always present, as far as we know.